### PR TITLE
Test for #12094

### PR DESCRIPTION
**Pull request by Jurigag**

### CODE REVIEW

File: tests/unit/Acl/Adapter/MemoryTest.php [hunk 1]
```diff
@@ -578,4 +577,50 @@ function () {
...
+
+                $acl->allow('Guests', 'Post', 'update', function($parameter) {
+                    return $parameter % 2 == 0;
+                });
+                $acl->allow('Members', 'Post', 'update', function($parameter) {
+                    return $parameter % 2 == 0;
+                });
+                $acl->allow('Admins', 'Post', 'update');
+                
+                expect(@$acl->isAllowed($guest, $model, 'update'))->false();
```
Comment (sergeyklay) [inline tests/unit/Acl/Adapter/MemoryTest.php]: Could you plese remove `@`

Comment (Jurigag) [inline tests/unit/Acl/Adapter/MemoryTest.php]: Memory adapter produces a warning when user will not provide any parameters to inform him about this, this warning is converted to exception in phpunit.

Comment (sergeyklay) [inline tests/unit/Acl/Adapter/MemoryTest.php]:
```php
/**
 * @expectedException \PHPUnit_Framework_Error_Warning
 */
```

Comment (Jurigag) [inline tests/unit/Acl/Adapter/MemoryTest.php]: Oh okay, where exactly add this ? On method or above expect ?

Comment (sergeyklay) [inline tests/unit/Acl/Adapter/MemoryTest.php]: Just split test to two tests - one with warnings, and second without. Add `@expectedException` annotation to the first

Comment (Jurigag) [inline tests/unit/Acl/Adapter/MemoryTest.php]: Oh okay. So keep @ and in second tests without @ and expectedException ? Okay.

Comment (sergeyklay) [inline tests/unit/Acl/Adapter/MemoryTest.php]: yeah

File: tests/unit/Acl/Adapter/MemoryTest.php [hunk 2]
```diff
@@ -578,4 +577,99 @@ function () {
...
+
+    /**
+     * Tests function in Acl Allow Method without arguments
+     *
+     * @issue   12094
+     *
+     * @author  Wojciech Slawski <jurigag@gmail.com>
+     * @since   2016-06-05
+     *
+     * @expectedException           PHPUnit_Framework_Exception
```
Comment (sergeyklay) [inline tests/unit/Acl/Adapter/MemoryTest.php]: Where exacltly throws PHPUnit_Framework_Exception? What `$acl->isAllowed` throws exception?
Could you please split this method to a few small tests?  Or just use `isAllowed` once?

Comment (Jurigag) [inline tests/unit/Acl/Adapter/MemoryTest.php]: In isAllowed method. Just `PHPUnit_Framework_Error_Warning` doesn't work, trigger_error is used in zephir source.

**Comment by sergeyklay** [comment]
Thanks

**Pull request merged**